pkg/app/context.go

```diff
@@ -78,20 +78,70 @@ type Handler interface {
 
 type ClientIP func(ctx *RequestContext) string
 
-var defaultClientIP = func(ctx *RequestContext) string {
-	RemoteIPHeaders := []string{"X-Real-IP", "X-Forwarded-For"}
-	for _, headerName := range RemoteIPHeaders {
-		ip := ctx.Request.Header.Get(headerName)
-		if ip != "" {
-			return ip
+type ClientIPOptions struct {
+	RemoteIPHeaders []string
+	TrustedProxies  map[string]bool
+}
+
+var defaultClientIPOptions = ClientIPOptions{
+	RemoteIPHeaders: []string{"X-Real-IP", "X-Forwarded-For"},
+	TrustedProxies: map[string]bool{
+		"0.0.0.0": true,
+	},
+}
+
+// SetClientIPOptionsFunc used to custom ClientIP-related configuration
+func SetClientIPOptionsFunc(opts ClientIPOptions) {
+	defaultClientIPOptions = opts
```

Comment:
This will change the global options for all instance - this may cause some problems - perhaps binding this to the hertz instance?

Reply:
`defaultClientIP` is also a global function too. If someone want to binding to a hertz instance, I perfer to use `ctx.SetClientIPFunc`

Reply:
no and refer to  https://github.com/cloudwego/hertz/pull/473

Reply:
![image](https://user-images.githubusercontent.com/97824201/218132992-16895eb3-8c15-45cc-b869-b20ac0dc3e24.png)